listenbrainz/webserver/views/api_tools.py

```diff
@@ -104,6 +105,9 @@ def validate_listen(listen, listen_type):
             log_raise_400("JSON document may only contain listened_at and "
                            "track_metadata top level keys", listen)
 
+        if listen['listened_at'] > int(time.time()):
```

Comment:
perhaps it's a good idea to consider a bit of clock skew. I'd consider allowing anything from 10 minutes to 1h in advance